pkg/app/app.go

```diff
@@ -37,6 +38,72 @@ func defaultFunc_FF_CODE_CONSISTENT_IGNORES(engine workflow.Engine, config confi
 	return callback
 }
 
+func defaultFuncOrganization(engine workflow.Engine, config configuration.Configuration, apiClient api.ApiClient, logger *zerolog.Logger) configuration.DefaultValueFunction {
+	callback := func(existingValue interface{}) interface{} {
+		client := engine.GetNetworkAccess().GetHttpClient()
+		url := config.GetString(configuration.API_URL)
+		apiClient.Init(url, client)
+		existingString, ok := existingValue.(string)
+		if existingValue != nil && ok && len(existingString) > 0 {
+			orgId := existingString
+			_, err := uuid.Parse(orgId)
+			isSlugName := err != nil
+			if isSlugName {
+				orgId, err = apiClient.GetOrgIdFromSlug(existingString)
+				if err != nil {
+					logger.Print("Failed to determine default value for \"ORGANIZATION\":", err)
+				} else {
+					return orgId
+				}
+			} else {
+				return orgId
+			}
+		}
+
+		orgId, err := apiClient.GetDefaultOrgId()
+		if err != nil {
+			logger.Print("Failed to determine default value for \"ORGANIZATION\":", err)
+		}
+
+		return orgId
+	}
+	return callback
+}
+
+func defaultFuncApiUrl(logger *zerolog.Logger) configuration.DefaultValueFunction {
+	callback := func(existingValue interface{}) interface{} {
+		urlString := constants.SNYK_DEFAULT_API_URL
+
+		if existingValue != nil {
+			if temp, ok := existingValue.(string); ok {
+				urlString = temp
+			}
+		}
+
+		apiString, err := api.GetCanonicalApiUrlFromString(urlString)
+		if err != nil {
+			logger.Warn().Err(err).Str(configuration.API_URL, urlString).Msg("failed to get api url")
+		}
+		return apiString
+	}
+	return callback
+}
+
+func defaultInputDirectory() configuration.DefaultValueFunction {
+	callback := func(existingValue interface{}) interface{} {
```

Comment:
question: is this a go pattern? do we need to declare/assign `callback` in these methods if we just return it? Could we not just return the callback: `return func(existingValue interface{}) interface{} { ...`

Reply:
would be possible as well.